### PR TITLE
seems to generally work, use with most recent convoy branch

### PR DESCRIPTION
**Pull request by asutherland**
key disclaimers:
- I couldn't get js/sync.js to require('co') or variations that seemed like
  they should have worked like require('ext/ext/co') or require('gelam/ext/co')
  or others for unknown reasons, even with also updating the build paths.
- co at js/co makes jshint angry because it does not 'use strict' so I did this
  commit with -n to bypass that
- when running on mulet, it seems like overdue alarms get lost at startup, but
  I think that's an artifact of how mulet works.

### CODE REVIEW

**Comment by jrburke** [comment]
If you were trying to reuse the co in gelam, then if doing a build, we remove files in the ext directory that were part of a gelam layer, so I can see why co could not be located in that case.

**Comment by asutherland** [comment]
Ah, yes, that would explain things!  I was doing a build because DEBUG=0 still doesn't work for mulet for me.

File: apps/email/js/html_cache_restore.js [hunk 1]
```diff
@@ -32,17 +32,39 @@ window.HTML_CACHE_VERSION = '2';
...
+ * false but an 'alarm' is subsequently delivered during async startup.  This
+ * means that you can rely on the backend not having consumed an 'alarm' system
+ * message if this variable is false.
+ *
+ * Note that if the mozHasPendingMessage('alarm') was true, this variable will
+ * be initialized to 'alarm' as part of our startup process without the backend
+ * needing to be involved at all.
  */
-window.appDispatchedMessage = false;
+window.appShouldStayAlive = false;
```
Comment (jrburke) [inline apps/email/js/html_cache_restore.js]: This seems like it should be set to null to start.

What I will do is just start with merging this pull request as is, but I will comment in here for things I will change after doing the merge, so if there is a change that would be counter to the design goals you can comment on it. 

This is null setting is the first example of one of those changes.

Comment (asutherland) [inline apps/email/js/html_cache_restore.js]: Agreed.  This went through a few iterations as I thought about things and it was originally a boolean with another friend variable, but I later simplified it to just be the one variable but failed to update the initializing value.

I should probably also say this is really the only set of changes made to gaia mail where things were deeply thought through.  (Although I probably could have better done a better job on the name here.  Feel free to change it!)  Everything else I was just trying to duct tape stuff together to prove it works and then you can make things sane, etc.  And I filed bugs for the bigger discussion points we should probably have.  Although it'll probably be early next week before I'm really able to get traction on addressing any of those points.

**Pull request merged**